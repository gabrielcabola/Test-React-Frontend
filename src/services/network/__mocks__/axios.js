// ./__mocks__/axios.js
import mockAxios from 'jest-mock-axios';
export default mockAxios;

export const articles = {
  "articles": [
    {
      "manufacturer": "Yamaha",
      "model": "C40",
      "mainImage": {
        "fileName": "135317.jpg"
      }
    }]
};