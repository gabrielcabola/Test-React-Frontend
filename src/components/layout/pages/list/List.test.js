import React from "react";
import renderer from 'react-test-renderer';
import List from './';
import { articles } from 'services/network/__mocks__/axios';
import { BrowserRouter } from "react-router-dom";

it('renders correctly', () => {
  const tree = renderer
    .create(<BrowserRouter><List {...articles}  /></BrowserRouter>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
