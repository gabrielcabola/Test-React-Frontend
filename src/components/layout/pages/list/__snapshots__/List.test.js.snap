// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<section
  className="List"
>
  <article>
    <a
      className="articleLink"
      href="/article/0/135317.jpg"
      onClick={[Function]}
    >
      <div
        className="thumbImage"
      >
        <img
          src="https://thumbs.static-thomann.de/thumb/thumb120x/pics/prod//135317.jpg"
        />
      </div>
      <h2>
        C40
      </h2>
      <h3>
        Yamaha
      </h3>
    </a>
  </article>
</section>
`;
